source/base/parameter_handler.cc

```diff
@@ -612,6 +612,27 @@ namespace
           }
       }
   }
+
+  void
+  simplify_xml_recursively(boost::property_tree::ptree &source)
```

Comment:
This function doesn't actually have anything to do with XML -- it just works on the ptree. Maybe adjust the name of the function?